src/test/groovy/graphql/nadel/engine/OverallQueryTransformerTest.groovy

```diff
@@ -0,0 +1,150 @@
+package graphql.nadel.engine
+
+import graphql.execution.ExecutionContext
+import graphql.execution.nextgen.FieldSubSelection
+import graphql.language.AstPrinter
+import graphql.language.Document
+import graphql.language.Field
+import graphql.language.OperationDefinition
+import graphql.nadel.TestUtil
+import graphql.schema.GraphQLSchema
+import spock.lang.Specification
+
+import java.util.stream.Collectors
+
+class OverallQueryTransformerTest extends Specification {
+    def schema = TestUtil.schemaFromNdsl('''
+        service example {
+            type Query { 
+                hello: String <= $source.helloWorld 
+                foo(id: ID!): Foo
+                bar(id: ID!): Bar
+            }
+            
+            type Foo {
+                id: ID!
+                barId: String <= $source.bazId
+                qux: String!
+            }
+            
+            type Bar <= $innerTypes.Baz {
+                id: ID!
+            }
+        }
+         ''')
+
+    def "transforms query to delegate with field rename"() {
+        def query = TestUtil.parseQuery(
+                '''
+            {
+             hAlias: hello 
+             foo(id: "1") {
+                fooId: id
+                barId
+             }
+            }
+            ''')
+
+        when:
+        def delegateQuery = doTransform(schema, query)
+
+        then:
+        AstPrinter.printAstCompact(delegateQuery) == "query {hAlias:helloWorld foo(id:\"1\") {fooId:id bazId}}"
+    }
+
+    def "referenced named fragments are transformed and included"() {
+        def query = TestUtil.parseQuery(
+                '''
+            {
+             f1: foo(id: "1") {
+                ...frag1
+             }
+             f2: foo(id: "2") {
+                ...frag2
+             }
+            }
+            fragment frag1 on Foo {
+                id
+            }
+            fragment frag2 on Foo {
+                id
+                barId
+            }
+            fragment unrefFrag on Foo {
+                barId
+            }
+            ''')
+
+        when:
+        def delegateQuery = doTransform(schema, query)
+
+        then:
+        AstPrinter.printAstCompact(delegateQuery) ==
```

Comment:
Style nit - I really think static imports read much better - especially in tests